src/model/config.ts

```diff
@@ -0,0 +1,54 @@
+import BN from "bn.js";
+import { ConfigurableCorePool } from "../core/configurablePool";
+
+export class PoolConfig {
+    id: string;
+    tickSpacing: number;
+    token0: string;
+    token1: string;
+    fee: number;
+}
+
+export class Record {
+    id: string;
+    eventId: number;
+    actionName: string;
+    actionParams: object;
+    timestamp: Date;
+}
+
+export class Snapshot {
+    id: string;
+    nonce: number;
+    baseSnapshot: Snapshot;
+    poolConfig: PoolConfig;
+    records: Record[];
+    baseRecordId: string;
+    lastRecordId: string;
+    used: boolean;
```

Comment:
For this kind of data record structure, ideally, we should keep them immutable and stateless after being retrieved from the database. Adding state-tracking fields/properties tends to mislead developers to treat these objects as mutable and update non-state-tracking fields when working on the code.

This rule should apply to other data objects as well.

We could track any possible state in the higher-level code that is consuming the objects.

Reply:
There is a state structure immutable from the database, and also a structure tracking current simulator's running history. It seems now they are kind of mixed.

Reply:
Just to clarify, by "state", I meant in the sense of general programming, like "a state of an object", not the "state" in the simulator.

As a pure data object, these structures shouldn't have a mutable internal state, it's better to make them plain, immutable objects that are being passed around.